src/tracer.c

```diff
@@ -35,29 +35,36 @@
 #define min_not_zero(l, r) ((l) == 0) ? (r) : (((r) == 0) ? (l) : min(l, r))
 #endif
 
-static int blk_stack_limits(struct request_queue *t, struct request_queue *b,
+int blk_stack_limits(struct queue_limits *t, struct queue_limits *b,
                             sector_t offset)
 {
         t->max_sectors = min_not_zero(t->max_sectors, b->max_sectors);
         t->max_hw_sectors = min_not_zero(t->max_hw_sectors, b->max_hw_sectors);
-        t->bounce_pfn = min_not_zero(t->bounce_pfn, b->bounce_pfn);
+        t->bounce = min_not_zero(t->bounce, b->bounce);
         t->seg_boundary_mask =
                 min_not_zero(t->seg_boundary_mask, b->seg_boundary_mask);
-        t->max_phys_segments =
-                min_not_zero(t->max_phys_segments, b->max_phys_segments);
-        t->max_hw_segments =
-                min_not_zero(t->max_hw_segments, b->max_hw_segments);
+        t->max_segments =
+                min_not_zero(t->max_segments, b->max_segments);
+        t->max_segments =
+                min_not_zero(t->max_segments, b->max_segments);
         t->max_segment_size =
-                min_not_zero(t->max_segment_size, b->max_segment_size);
+                min_not_zero(t->max_segment_size, b->max_segment_size);     
         return 0;
 }
 
+static int blk_stack_limits_request_queue(struct request_queue *t, struct request_queue *b,
+                            sector_t offset)
```

Comment:
Do we need offset? It's not used.

Reply:
It is strange legacy - in obsolete kernel definition offset param was present: blk_stack_limits(struct request_queue *t, struct request_queue *b,
                            sector_t offset)
but not used in our implementation, even though the function was called:
 return blk_stack_limits(t, bq, start << 9);
Same happens for our new definition with param type request_queue changed to queue_limits. I kept this offset to keep compliance with the call from line 67 although I don't know why we keep using start << 9 there.